torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -244,6 +244,7 @@ void ProcessGroupAgent::start() {
 void ProcessGroupAgent::shutdown() {
   LOG(INFO) << "Shutting down ProcessGroupAgent on rank " << pg_->getRank()
             << ".";
+  RpcAgent::shutdown();
```

Comment:
Wondering: does it make more sense to put the superclass's shutdown() method at the end?
This looks functionally like a destructor idiom, in which case one normally expects the subclass to destruct itself before deferring to the superclass's shutdown.

Reply:
(My comment also applies in parallel to ThriftRpcAgent)

Reply:
Sounds good, I'll make this change